src/affprop.jl

```diff
@@ -21,16 +30,40 @@ const _afp_default_damp = 0.5
 const _afp_default_tol = 1.0e-6
 const _afp_default_display = :none
 
-function affinityprop(S::DenseMatrix{T}; 
+"""
+    affinityprop(S::DenseMatrix; [maxiter=200], [tol=1e-6], [damp=0.5],
+                 [display=:none])
+
+Perform affinity propagation clustering based on a similarity matrix `S`.
+
+``S_{ij}`` (``i ≠ j``) is the similarity (or the negated distance) between
+the ``i``-th and ``j``-th points, ``S_{ii}`` defines the *availability*
+of the ``i``-th point as an *exemplar*.
+
+Returns an instance of [`AffinityPropResult`](@ref).
```

Comment:
Minor point: in the Julia manual, the convention is to use the imperative "return". Same elsewhere, including for "perform" and so on.